src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,131 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+    
+    private static class Node<T> {
+        private T data;
+        private Node<T> next;
+        private Node<T> prev;
+        
+        private Node(Node<T> prev, T data, Node<T> next) {
+            this.prev = prev;
+            this.data = data;
+            this.next = next;
+        }
+    }
+    
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(tail, value, null);
+        if (tail == null) {
+            head = newNode;
+        } else {
+            tail.next = newNode;
+        }
+        tail = newNode;
+        size++;
     }
-
+    
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+        } else {
+            Node<T> node = getNode(index);
+            Node<T> newNode = new Node<>(node.prev, value, node);
+            if (node.prev != null) {
+                node.prev.next = newNode;
+            } else {
+                head = newNode;
+            }
+            node.prev = newNode;
+            size++;
+        }
     }
-
+    
     @Override
     public void addAll(List<T> list) {
+        for (T value : list) {
+            add(value);
+        }
     }
-
+    
     @Override
     public T get(int index) {
-        return null;
+        return getNode(index).data;
     }
-
+    
     @Override
     public T set(T value, int index) {
-        return null;
+        Node<T> node = getNode(index);
+        T oldValue = node.data;
+        node.data = value;
+        return oldValue;
     }
-
+    
     @Override
     public T remove(int index) {
-        return null;
+        Node<T> node = getNode(index);
+        unlink(node);
+        size--;
+        return node.data;
     }
-
+    
     @Override
     public boolean remove(T object) {
+        for (Node<T> node = head; node != null; node = node.next) {
+            if (object == null ? node.data == null : object.equals(node.data)) {
+                unlink(node);
+                size--;
+                return true;
+            }
+        }
         return false;
     }
-
+    
     @Override
     public int size() {
-        return 0;
+        return size;
     }
-
+    
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+    
+    private Node<T> getNode(int index) {
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException("Index" + index + "out of bounds exception");
```

Comment:
Also show current size in this message
```suggestion
            throw new IndexOutOfBoundsException("Index: " + index + " out of bounds exception");
```